src/librustc_span/lib.rs

```diff
@@ -888,6 +892,36 @@ impl ExternalSource {
 #[derive(Debug)]
 pub struct OffsetOverflowError;
 
+#[derive(Copy, Clone, RustcEncodable, RustcDecodable)]
+pub enum SourceFileChecksumKind {
+    Md5,
+    Sha1,
+}
+
+/// The hash of the on-disk source file used for debug info.
+#[derive(Clone, RustcEncodable, RustcDecodable)]
+pub struct SourceFileChecksum {
+    pub kind: SourceFileChecksumKind,
+    value: Vec<u8>,
```

Comment:
Would be nice to use a fixed-size array here, or a pair of `u128` and `u32` (for SHA-1).